pkg/install/generate.go

```diff
@@ -615,3 +648,64 @@ func completeOpenStackDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv
 	job.Spec.Template.Spec.Containers = containers
 	job.Spec.Template.Spec.Volumes = volumes
 }
+
+func completeVSphereDeprovisionJob(req *hivev1.ClusterDeprovision, job *batchv1.Job) {
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	credentialsSecret := ""
+	if len(req.Spec.Platform.VSphere.CredentialsSecretRef.Name) > 0 {
```

Comment:
I don't think that we can skip adding the env variable if there is no secret specified.